pkg/webhook/vlanconfig/validator.go

```diff
@@ -143,6 +144,17 @@ func (v *Validator) Delete(_ *admission.Request, oldObj runtime.Object) error {
 		return fmt.Errorf(deleteErr, vc.Name, err)
 	}
 
+	nads, err := v.nadCache.List(StorageNetworkNetAttachDefNamespace, labels.Set(map[string]string{
+		utils.KeyClusterNetworkLabel: vc.Spec.ClusterNetwork,
+	}).AsSelector())
+	if err != nil {
+		return fmt.Errorf(deleteErr, vc.Name, err)
+	}
+
+	if len(nads) > 0 {
+		return fmt.Errorf(deleteErr, vc.Name, fmt.Errorf("storage network is still attached"))
```

Comment:
Here we filter NADs only with the CN key. Stating that "storage network is still attached" feels weird. We can't be so sure it's a "storage" network. Technically, Harvester users can create NADs under the `harvester-system` NS. When those VCs get deleted with their associated NADs still existing, the webhook will show such a message, which is probably inaccurate.

Reply:
Not sure if there are use cases where users could create NAD via other methods which creates nads in harvester-system namespace.Our harvester by design creating all other non storage network nads only in default namespace.But its good not to assume, we could check the nad's name prefix for "storagenetwork" but to be even stricter, storage network's nad's annotation field is updated specifically to identify its as storage network nad during creation.
So I have used that to identify the nad specifically as "storage network" nad.